src/metabase/models/dashboard.clj

```diff
@@ -183,36 +183,62 @@
   (-> dashboard
       (select-keys [:collection_id :description :name :cache_ttl :auto_apply_filters])
       (assoc :cards (vec (for [dashboard-card (ordered-cards dashboard)]
-                           (-> (select-keys dashboard-card [:size_x :size_y :row :col :id :card_id])
-                               (assoc :series (mapv :id (dashboard-card/series dashboard-card)))))))))
+                           (-> (select-keys dashboard-card [:dashboard_tab_id :size_x :size_y :row :col :id :card_id])
+                               (assoc :series (mapv :id (dashboard-card/series dashboard-card)))))))
+      (assoc :tabs (map #(select-keys % [:id :name :position :dashboard_id]) (ordered-tabs dashboard)))))
+
+(defn- revert-tabs
```

Comment:
The bulk of this function looks suspiciously like `metabase.api.dashboard/do-update-tabs!`. I think you can do a refactor to this function so it takes a dashboard-id instead of a dashboard, and move it to the `metabase.models.dashboard-tab` namespace. Then you can do something like this:

```clojure 
(defn- revert-tabs
  [dashboard-id serialized-tabs]
  (let [current-tabs             (t2/select-fn-vec #(dissoc :created_at :updated_at :entity_id) :model/DashboardTab
                                                   :dashboard_id dashboard-id)
        {:keys [old-id->new-id]} (dashboard-tab/do-update-tabs! dashboard-id current-tabs serialized-tabs)]
    {:old-id->new-id old-id->new-id}))
```

Reply:
moved it to `do-updated-tabs` and reused both in API and revert